### PR TITLE
add cryptopunks_trade

### PR DESCRIPTION
**Pull request by Jan-zou**
add cryptopunks_trades
modify floor price in smart nft holder label

### CODE REVIEW

File: models/nft/cryptopunks_trades.sql [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+  null as aggregator,
+  case when b.num > 1 then 'Bundle Trade' else 'Single Item Trade' end as trade_type,
+  a.from_address as buyer,
+  a.to_address as seller,
+  'CryptoPunks' as nft_project_name,
+  a.original_currency_amount / power(10, 18) as currency_amount,
+  'ETH' as currency_symbol,
+  '0xc02aaa39b223fe8d0a0e5c4f27ead9083c756cc2' as currency_contract,
+  null as usd_amount, -- a.original_currency_amount/power(10,18)*p.price as usd_amount
+  a.original_currency_amount / power(10, 18) as eth_amount,
```
[the remark this answers is not in the document]
Comment (onesuper) [reply models/nft/cryptopunks_trades.sql:85]: I noticed the usd_amount calculation logic is masked off.

[the remark this answers is not in the document]
Comment (Jan-zou) [reply models/nft/cryptopunks_trades.sql:87]: I found that there is no record of CryptoPunks in `prices.usd` table.

File: models/nft/cryptopunks_trades.sql [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+  from (
+    select
+      a.*,
+      b.value as bid_value,
+      row_number()over(partition by a.nft_token_id, a.block_time order by b.evt_block_time desc) as rank
+    from address_info a
+    left join cryptopunksmarket_evt_punkbidentered b
+      on a.nft_token_id = b.punkIndex and b.fromAddress = a.to_address and b.evt_block_time <= a.block_time
+  )
+  where rank = 1 and value < 20000 * power(10, 18)
```
Comment (onesuper) [inline models/nft/cryptopunks_trades.sql:58]: What's the meaning of 20000?

Comment (Jan-zou) [reply models/nft/cryptopunks_trades.sql:58]: Many dune dashboards add this filter, I found just filtered the transaction of #9998 (124457ETH, the money from this NFT trade ended up right back where it started）。
Maybe used in the dashboard to filter outliers,  I'd remove this filter here.

Comment (onesuper) [reply models/nft/cryptopunks_trades.sql]: OKay. I guess the filter is reasonable. Maybe we can just filter by transaction id and add some comments to it.

**Pull request merged**